panic: runtime error: invalid memory address or nil pointer dereference

**What did you do?**
Restarted prometheus

**What did you expect to see?**
Started prometheus
**What did you see instead? Under which circumstances?**

**Environment**

Linux

* System information:
```
Linux 4.4.0-38-generic x86_64
```

* Prometheus version: 
```
prometheus, version 1.3.0 (branch: master, revision: 18254a172b1e981ed593442b2259bd63617d6aca)
  build user:       root@d363f050a0e0
  build date:       20161101-17:06:27
  go version:       go1.7.3
```
* Prometheus configuration file:
```
insert configuration here
```

* Logs:
```
INFO[0726] All requests for rebuilding the label indexes queued. (Actual processing may lag behind.)  source=crashrecovery.go:528
INFO[0726] Checkpointing fingerprint mappings...         source=persistence.go:1372
INFO[0726] Done checkpointing fingerprint mappings in 99.016883ms.  source=persistence.go:1395
WARN[0726] Crash recovery complete.                      source=crashrecovery.go:151
INFO[0726] 98806 series loaded.                          source=storage.go:359
INFO[0726] Starting target manager...                    source=targetmanager.go:76
INFO[0726] Listening on :9090                            source=web.go:240
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x637b3a]

goroutine 275 [running]:
panic(0x1764a00, 0xc4200140a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
regexp.(*Regexp).FindStringSubmatchIndex(0x0, 0x0, 0x0, 0x7, 0x1, 0x0)
	/usr/local/go/src/regexp/regexp.go:922 +0x7a
github.com/prometheus/prometheus/relabel.relabel(0xc44612d440, 0xc420037a80, 0xc44612d440)
	/go/src/github.com/prometheus/prometheus/relabel/relabel.go:59 +0x6e3
github.com/prometheus/prometheus/relabel.Process(0xc420064e40, 0xc420483620, 0x3, 0x3, 0x6)
	/go/src/github.com/prometheus/prometheus/relabel/relabel.go:35 +0x158
github.com/prometheus/prometheus/retrieval.populateLabels(0xc420064e40, 0xc4204d4480, 0xc481be8c70, 0xc476d043a8, 0x0, 0x1)
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:461 +0x39e
github.com/prometheus/prometheus/retrieval.targetsFromGroup(0xc420064c60, 0xc4204d4480, 0x0, 0xc4204e6d00, 0x1, 0x0, 0xc495381130)
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:523 +0x1b1
github.com/prometheus/prometheus/retrieval.(*targetSet).runProviders.func1(0xc452d13510, 0xc498ebbec0, 0xc495381a40, 0xc452d13520, 0xc452d13508, 0x8, 0x2426580, 0xc45741ece0)
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:298 +0x4d6
created by github.com/prometheus/prometheus/retrieval.(*targetSet).runProviders
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:330 +0x2dc
```

--- a/config/config.go
+++ b/config/config.go
@@ -1017,6 +1017,9 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := checkOverflow(c.XXX, "relabel_config"); err != nil {
 		return err
 	}
+	if c.Regex.Regexp == nil {
+		c.Regex = MustNewRegexp("")
+	}
 	if c.Modulus == 0 && c.Action == RelabelHashMod {
 		return fmt.Errorf("relabel configuration for hashmod requires non-zero modulus")
 	}

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -115,6 +115,12 @@ var expectedConf = &Config{
 					Regex:       DefaultRelabelConfig.Regex,
 					Replacement: "static",
 					Action:      RelabelReplace,
+				}, {
+					TargetLabel: "abc",
+					Separator:   ";",
+					Regex:       MustNewRegexp(""),
+					Replacement: "static",
+					Action:      RelabelReplace,
 				},
 			},
 		},

--- a/config/testdata/conf.good.yml
+++ b/config/testdata/conf.good.yml
@@ -54,6 +54,9 @@ scrape_configs:
     target_label:  cde
   - replacement:   static
     target_label:  abc
+  - regex:
+    replacement:   static
+    target_label:  abc
 
   bearer_token_file: valid_token_file
 

